core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -307,6 +308,15 @@ default Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessi
                 .iterator();
     }
 
+    /**
+     * Gets the metadata for all columns that match the specified table prefix. Redirected table names are included, but the column metadata for them is not. Additional
+     * <code>tableFilter</code> parameter can be used to prune tables from column search.
+     */
+    default Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession session, SchemaTablePrefix prefix, UnaryOperator<Set<SchemaTableName>> tablesFilter)
```

Comment:
Going further, we could allow the connector to fetch column information for MVs, views and tables in one call. This would let avoid listing relations 3 times, when compared to `getMaterializedViews + getViews + streamTableColumns` currently used to satisfy `information_schema.columns`.
I drafted a PR showing how this could look like: https://github.com/trinodb/trino/pull/18585